lib/Service/DisplayNotifyService.php

```diff
@@ -219,13 +219,15 @@ public function notifyByDisplayGroupId($displayGroupId)
     }
 
     /** @inheritdoc */
-    public function notifyByCampaignId($campaignId)
+    public function notifyByCampaignId($campaignIds)
```

Comment:
Sorry, but having looked at this I don't think I like passing round the campaignIds.

I think it would be better if this method went back to how it was, and we implemented the full SQL in the `notifyByLayoutCode` method.

Even though that is more code and repeated code, what actually runs is clean, parameterised and memory efficient.